polaris.shopify.com/src/types.ts

```diff
@@ -24,3 +24,14 @@ export type LineConfig = {
   fromOrigin?: "center" | "edge" | "left" | "right";
   toOrigin?: "center" | "edge" | "left" | "right";
 };
+
+interface Icon {
```

Comment:
Thoughts on calling this `IconMetadata` since it relates to the icon info rather than the icon resource itself? If we want to support metadata from the npm package, the type information should probably come from there.

To @BPScott's point, perhaps we should export the metadata as JS (but with TS declarations). That would improve import/require compatibility, include type info, and consumers could easily convert to JSON if desired with `JSON.parse(meta)`. Then we could also probably remove the export key in the `package.json` and instead just export metadata from the package index file.

Reply:
> Then we could also probably remove the export key in the package.json and instead just export metadata from the package index file.

I'd probably lean towards keeping a separate import for the metadata as it isn't inntended to be used outside of our documentation. Rather than make bundler dead-code-eliminate that big data object, I'd lean towards not including it in the main entrypoint in the first place